mlflow/store/dbmodels/models.py

```diff
@@ -45,11 +45,12 @@ def _create_entity(base, model):
                     metrics = {}
                     for o in obj:
                         existing_metric = metrics.get(o.key)
-                        if (existing_metric is None) or (o.timestamp > existing_metric.timestamp)\
-                            or (o.timestamp == existing_metric.timestamp
-                                and o.value > existing_metric.value):
-                            metrics[o.key] = Metric(o.key, o.value, o.timestamp)
-                    obj = metrics.values()
+                        if (existing_metric is None)\
+                            or ((o.step, o.timestamp, o.value) >=
```

Comment:
hypernit, but I think `>` is a tiny bit more logical, since if it's equal we don't need to do anything